tests/cocotb/test_snitch_cc.py

```diff
@@ -0,0 +1,89 @@
+# ---------------------------------
+# Copyright 2023 KULeuven
+# Solderpad Hardware License, Version 0.51, see LICENSE for details.
+# SPDX-License-Identifier: SHL-0.51
+# Author: Ryan Antonio (ryan.antonio@esat.kuleuven.be)
+# ---------------------------------
+
+import cocotb
+from cocotb.triggers import RisingEdge
+from cocotb.clock import Clock
+from cocotb_test.simulator import run
+import pytest
+import snax_util
+
+
+# Testing parameters
+CLOCK_CYCLES = 20
+
+
+@cocotb.test()
+async def snitch_cc_dut(dut):
+    clock = Clock(dut.clk_i, 10, units="ns")
+    cocotb.start_soon(clock.start())
+
+    dut.rst_ni.value = 0
+
+    await RisingEdge(dut.clk_i)
+
+    dut.rst_ni.value = 1
+
+    for i in range(CLOCK_CYCLES):
+        # TODO: Fix me to be working later
+        # Simple check if instructions are running normally
+        # instruction_value = hex(dut.i_snitch_cc.i_snitch.inst_data_i.value)
+        # instruction_address = int(dut.instruction_addr_offset.value)
+
+        # cocotb.log.info('---------- Instruction Info ----------')
+        # cocotb.log.info(f'Instruction Value: {instruction_value}')
+        # cocotb.log.info(f'Instruction Addr: {instruction_address}')
+
+        await RisingEdge(dut.clk_i)
+
+
+# Main test run
+@pytest.mark.parametrize("parameters", [{"AddrWidth": str(48), "DataWidth": str(64)}])
+def test_snitch_cc(parameters, simulator):
+    tests_path = "./tests/cocotb/"
+
+    includes, defines, verilog_sources = snax_util.extract_bender_filelist()
+
```

Comment:
Please refactor the shift register cocotb python file in the same way

Reply:
Done! I had to make an additional utility function that extracts specific module file paths only.